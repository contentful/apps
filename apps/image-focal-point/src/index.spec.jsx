import React from 'react';
import { App } from './index';
import { render, cleanup, configure } from '@testing-library/react';
import { vi } from 'vitest';

import mockProps from './test/mockProps';

const sdk = {
  ...mockProps.sdk,
  field: {
    getValue: vi.fn(),
    onValueChanged: vi.fn(),
    setValue: vi.fn(),
    removeValue: vi.fn(),
  },
  window: {
    startAutoResizer: vi.fn(),
  },
};

vi.mock('./utils', () => ({
  getField: vi.fn(),
  isCompatibleImageField: () => true,
}));

configure({
  testIdAttribute: 'data-test-id',
});

function renderComponent(sdk) {
  return render(<App sdk={sdk} />);
}

describe('App', () => {
  beforeEach(() => {
    vi.resetAllMocks();
  });

  afterEach(cleanup);

  it('should read a value from field.getValue() and subscribe for external changes', () => {
    const initialValue = 'x: 0px / y: 0px';
    sdk.field.getValue.mockImplementation(() => initialValue);
    const { getByTestId } = renderComponent(sdk);

    expect(sdk.field.getValue).toHaveBeenCalled();
    expect(sdk.field.onValueChanged).toHaveBeenCalled();
    expect(getByTestId('focal-point').value).toEqual(initialValue);
  });

  it('should call startAutoResizer', () => {
    renderComponent(sdk);
    expect(sdk.window.startAutoResizer).toHaveBeenCalled();
  });

  describe('#render', () => {
    it('should render the extension field view', () => {
      const { container } = renderComponent(sdk);
      expect(container).toMatchSnapshot();
    });
  });
});
