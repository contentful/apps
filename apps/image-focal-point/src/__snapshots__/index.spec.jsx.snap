// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > #render > should render the extension field view 1`] = `
<div>
  <div
    class="css-70qvj9"
  >
    <input
      aria-describedby="focal-point-helptext"
      class="css-82ji86 css-93644d"
      data-test-id="focal-point"
      id="focal-point"
      type="text"
      value="Focal point not set"
      width="large"
    />
    <button
      buttontype="muted"
      class="css-ndrb9n css-dp0f2t"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="css-40g50j"
      >
        Set focal point
      </span>
    </button>
    <a
      class="css-1wm2dh8"
      data-test-id="cf-ui-text-link"
      linktype="primary"
    >
      <span
        class="css-x1sij0"
      >
        <span>
          Reset focal point
        </span>
      </span>
    </a>
  </div>
</div>
`;
